frontend/src/scenes/data-warehouse/viewLinkLogic.tsx

```diff
@@ -363,27 +369,28 @@ export const viewLinkLogic = kea<viewLinkLogicType>([
             await loadTablePreviewData(tableName, actions.setJoiningTablePreviewData)
         },
         validateJoin: async () => {
+            const sourceTableKey = values.viewLink.source_table_key
+            const joiningTableKey = values.viewLink.joining_table_key
+
             if (
                 !values.selectedSourceTableName ||
                 !values.selectedJoiningTableName ||
-                !values.selectedSourceKey ||
-                !values.selectedJoiningKey
+                !sourceTableKey ||
+                !joiningTableKey
             ) {
                 actions.setIsJoinValid(false)
                 return
             }
             try {
-                const sourceTable = hogql.identifier(values.selectedSourceTableName)
-                const sourceKey = hogql.identifier(values.selectedSourceKey)
-                const joiningTable = hogql.identifier(values.selectedJoiningTableName)
-                const joiningKey = hogql.identifier(values.selectedJoiningKey)
+                const sourceKeyExpression = `${values.selectedSourceTableName}.${sourceTableKey}`
+                const joiningKeyExpression = `${values.selectedJoiningTableName}.${joiningTableKey}`
```

Comment:
btw, a join expression can be a `ast.Call` instead of a `ast.Field` - e.g. `toString(properties.email)` - which is valid and won't work with the join validation stuff here